cmd/openshift-install/migrate/azure/migrate.go

```diff
@@ -0,0 +1,47 @@
+package azure
+
+import (
+	"github.com/spf13/cobra"
+
+	azmigrate "github.com/openshift/installer/pkg/migrate/azure"
+)
+
+var (
+	azureMigrateOpts struct {
+		zone              string
+		oldResourceGroup  string
+		newResourceGroup  string
+		virtualNetwork    string
+		vnetResourceGroup string
+		link              bool
+		public            bool
+	}
+)
+
+func runMigrateAzurePrivateDNSMigrateCmd(cmd *cobra.Command, args []string) error {
+	if (azureMigrateOpts.zone == "" || azureMigrateOpts.oldResourceGroup == "") || (azureMigrateOpts.link == true && azureMigrateOpts.virtualNetwork == "") || (azureMigrateOpts.virtualNetwork != "" && azureMigrateOpts.vnetResourceGroup == "") {
+		return cmd.Help()
+	}
+
+	return azmigrate.Migrate(azureMigrateOpts.oldResourceGroup, azureMigrateOpts.newResourceGroup, azureMigrateOpts.zone, azureMigrateOpts.virtualNetwork, azureMigrateOpts.vnetResourceGroup, azureMigrateOpts.link, azureMigrateOpts.public)
+}
+
+// NewMigrateAzurePrivateDNSMigrateCmd adds the migrate command to openshift-install
+func NewMigrateAzurePrivateDNSMigrateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Migrate a zone",
+		Long:  "This will migrate a legacy private zone to a new style private zone.",
+		RunE:  runMigrateAzurePrivateDNSMigrateCmd,
+	}
+
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.zone, "zone", "", "The zone to migrate")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.oldResourceGroup, "oldResourceGroup", "", "The resource group of the legacy zone")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.newResourceGroup, "newResourceGroup", "", "The resource group to create the private zone in")
```

Comment:
this doesn't seem like it's going to useful for openshift clusters.

Reply:
Can we guarantee that the legacy style private domain and the new style private domain will always be in the same resource group? Or can we make that a requirement? These options exist so that they can be different. If a new resource group is not specified, the old resource group is the default. Perhaps that could be made more clear.

Reply:
in the context of the openshift clusters, both the private dns and dns private resources will be in same resource group.

the vnet can be different resource group so having a flag for that is nice..., but defaulting it to the cluster resource group will be helpful.

Reply:
I've removed the options for new/old resource group. Now, just the resource group is required.